org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/templates/XtextTemplateContext.java

```diff
@@ -88,37 +87,40 @@ protected TemplateTranslator createTemplateTranslator() {
 			int offset = getStart();
 			IRegion lineRegion = getDocument().getLineInformationOfOffset(offset);
 			String line = getDocument().get(lineRegion.getOffset(), lineRegion.getLength());
+			String lineDelimiter = getDocument().getLineDelimiter(0);
 			int i = 0;
-			while(i < line.length() && Character.isWhitespace(line.charAt(i))) {
+			while (i < line.length() && Character.isWhitespace(line.charAt(i))) {
 				i++;
 			}
 			if (i != 0)
-				return new IndentationAwareTemplateTranslator(line.substring(0, i));
-			return new TemplateTranslator();
-		} catch(BadLocationException ex) {
+				return new IndentationAwareTemplateTranslator(line.substring(0, i), lineDelimiter);
+			return new IndentationAwareTemplateTranslator("", lineDelimiter);
+		} catch (BadLocationException ex) {
 			return new TemplateTranslator();
 		}
 	}
-	
+
 	public static class IndentationAwareTemplateTranslator extends TemplateTranslator {
-	
+
 		private final String indentation;
+		private final String lineDelimiter;
 
-		public IndentationAwareTemplateTranslator(String indentation) {
+		public IndentationAwareTemplateTranslator(String indentation, String lineDelimiter) {
 			this.indentation = indentation;
+			this.lineDelimiter = lineDelimiter;
 		}
 
 		@Override
 		public TemplateBuffer translate(Template template) throws TemplateException {
 			return translate(template.getPattern());
 		}
-		
+
 		@Override
 		public TemplateBuffer translate(String string) throws TemplateException {
 			String withIndentation = string.replaceAll("(\r\n?)|(\n)", "$0" + indentation);
```

Comment:
"$0" is the original line delimiter. Maybe you simply want to replace it by the line delimiter as passed into the constructor?

Reply:
Done.